src/Compilers/CSharp/Portable/Syntax/LookupPosition.cs

```diff
@@ -289,24 +289,9 @@ private static SyntaxToken GetFirstIncludedToken(StatementSyntax statement, bool
                 case SyntaxKind.FixedStatement:
                     return ((FixedStatementSyntax)statement).FixedKeyword;
                 case SyntaxKind.ForEachStatement:
-                    // NB: iteration variable is only in scope in body.
-                    ForEachStatementSyntax forEachSyntax = (ForEachStatementSyntax)statement;
-                    if (inRecursiveCall)
-                    {
-                        return forEachSyntax.ForEachKeyword;
-                    }
-                    return GetFirstIncludedToken(forEachSyntax.Statement, inRecursiveCall: true);
+                    return ((ForEachStatementSyntax)statement).OpenParenToken.GetNextToken();
```

Comment:
As we discussed offline, this code doesn't help to decide which scope we are in, it helps to decide if we are in a scope owned by a statement. The final scope determination is done by SemanticModel when it looks for the appropriate binder. AdjustBinderForPositionWithinStatement (in MemberSemanticModel.cs) specifically has compensating code that should address your concern.